src/components/fx/hover.js

```diff
@@ -950,6 +952,12 @@ function createHoverText(hoverData, opts, gd) {
             text = name;
         }
 
+        // hovertemplate
+        var trace = d.trace, hovertemplate = opts.hovertemplate || trace.hovertemplate || false;
+        if(hovertemplate) {
+            text = Lib.templateString(hovertemplate, gd._hoverdata[curveNumber], trace);
```

Comment:
We can pass additional objects as argument to `Lib.templateString` so the sky is the limit in terms of what values we want to make accessible here!

However, as mentioned in point 1, we should probably put this data into `_hoverdata` since it's emitted on hover.